haystack/schema.py

```diff
@@ -88,6 +88,14 @@ def __init__(
                              In this case, the id is generated by using the content and the defined metadata.
                              If you specify a custom ID for the `id` parameter, the `id_hash_keys` parameter is
                              ignored and the custom ID is used.
+
+                             Note that one can use even nested fields of the "meta" as id_hash_keys. For example, if you
+                             have a key in meta called "url" and you want to use it as part of the id, you can pass
+                             this parameter as ["meta.url"]. This will use the value of the "url" key in the "meta" dict.
+                             A maximum depth of 1 is supported. For example, if "meta.url.path" is used, it will look for
```

Comment:
can we make it active -> Haystack supports a max depth of 1.

and the others too + get rid of the future tense:
...if you use `meta.url.path`, it looks for..